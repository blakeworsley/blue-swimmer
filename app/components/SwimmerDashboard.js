'use strict';

import React, { Component } from 'react';
import { StyleSheet, Text, View, TextInput }  from 'react-native';
const styles = require('../styles.js');
const constants = styles.constants;

class SwimmerDashboard extends Component {
  constructor(props) {
     super(props);
     this.state = {
       physical: null,
       mental: null,
       effort: null,
       date: Date.now()
     };
   }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.swimmerDashboard}>{this.props.title}</Text>
          <Text>Physical</Text>
          <TextInput
            ref="1"
            style={styles.newUserInput}
            onChangeText={(raiting) => this.setState({raiting})}
            value={this.state.physical}
            placeholder="Rating between 1 and 5"
            keyboardType="numeric"
            returnKeyType="next"
            onSubmitEditing={() => this.focusNextField('2')}
            autoCapitalize="none"
          />
          <Text>Mental</Text>
          <TextInput
            ref="2"
            style={styles.newUserInput}
            onChangeText={(raiting) => this.setState({raiting})}
            value={this.state.mental}
            placeholder="Rating between 1 and 5"
            keyboardType="numeric"
            returnKeyType="next"
            onSubmitEditing={() => this.focusNextField('3')}
            autoCapitalize="none"
          />
          <Text>Effort</Text>
          <TextInput
            ref="2"
            style={styles.newUserInput}
            onChangeText={(raiting) => this.setState({raiting})}
            value={this.state.effort}
            placeholder="Rating between 1 and 5"
            keyboardType="numeric"
            returnKeyType="done"
            onSubmitEditing={() => this.focusNextField('submit')}
            autoCapitalize="none"
          />
          <TouchableHighlight
            ref="submit"
            style={styles.button}
          >
          <Text>Submit</Text>
          </TouchableHighlight>
      </View>
    );
  }
}

module.exports = SwimmerDashboard;
