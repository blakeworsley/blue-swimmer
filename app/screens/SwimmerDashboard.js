'use strict';

import firebase from '../firebase';
import React, { Component } from 'react';
import { StyleSheet, Text, View, TextInput, TouchableHighlight }  from 'react-native';
import split from 'split-object';
const styles = require('../styles.js');
const constants = styles.constants;

class SwimmerDashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      physical: null,
      mental: null,
      performance: null,
      date: Date.now(),
      user: null
    };
  }

  componentDidMount() {
    firebase.database().ref('users').on('value', (snapshot) => {
      let users = snapshot.val();
      split(users).map(user => {
        if (user.value.emailAddress === this.props.user.email) {
          users = Object.assign({ id: user.key }, user.value);
        }
      });
      this.setState({user: users});
    });
  }

  sendData() {
    const { physical, mental, performance, date, user } = this.state;
    firebase.database().ref(`workouts/${user.teamName}/${user.id}/${date}`).push({
      physical,
      mental,
      performance
    });
    // .then(() => { this.goToComplete(); });
  }

  focusNextField(nextField){
    this.refs[nextField].focus();
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.swimmerDashboard}>{this.props.title}</Text>
        <Text style={styles.h1Font}>{this.state.user ? 'Welcome ' + this.state.user.firstName : null}</Text>
        <Text style={styles.regularFont}>Physical</Text>
        <TextInput
          ref="1"
          style={styles.newUserInput}
          onChangeText={(rating) => this.setState({physical: rating})}
          value={this.state.physical}
          placeholder="Rating between 1 and 5"
          keyboardType="numeric"
          returnKeyType="next"
          onSubmitEditing={() => this.focusNextField('2')}
          autoCapitalize="none"
        />
        <Text style={styles.regularFont}>Mental</Text>
        <TextInput
          ref="2"
          style={styles.newUserInput}
          onChangeText={(rating) => this.setState({mental: rating})}
          value={this.state.mental}
          placeholder="Rating between 1 and 5"
          keyboardType="numeric"
          returnKeyType="next"
          onSubmitEditing={() => this.focusNextField('3')}
          autoCapitalize="none"
        />
        <Text style={styles.regularFont}>Effort</Text>
        <TextInput
          ref="2"
          style={styles.newUserInput}
          onChangeText={(rating) => this.setState({performance: rating})}
          value={this.state.performance}
          placeholder="Rating between 1 and 5"
          keyboardType="numeric"
          returnKeyType="done"
          onSubmitEditing={() => this.focusNextField('submit')}
          autoCapitalize="none"
        />
        <TouchableHighlight
          ref="submit"
          style={styles.button}
          onPress={() => {this.sendData()}}
        >
          <Text>Submit</Text>
        </TouchableHighlight>
        <TouchableHighlight
          style={styles.button}
          onPress={() => {
            firebase.auth().signOut();
          }}
        >
          <Text>Sign Out</Text>
        </TouchableHighlight>

        <Text style={styles.regularFont}>Physical: {this.state.physical || 'N/A'}</Text>
        <Text style={styles.regularFont}>Mental: {this.state.mental || 'N/A'}</Text>
        <Text style={styles.regularFont}>Performance: {this.state.performance || 'N/A'}</Text>
        <Text style={styles.regularFont}>Date: {this.state.date}</Text>
        <Text style={styles.regularFont}>Current User: {this.props.user.email}</Text>
        {this.state.user ? <Text style={styles.regularFont}>Team: {this.state.user.teamName}</Text> : null}


      </View>
    );
  }
}

module.exports = SwimmerDashboard;
