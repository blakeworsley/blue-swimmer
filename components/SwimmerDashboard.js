'use strict';

import React, { Component } from 'react';
import ReactNative from 'react-native';
const styles = require('../styles.js');
const constants = styles.constants;
const { StyleSheet, Text, View } = ReactNative;

class SwimmerDashboard extends Component {
  render() {
    return (
      <View style={styles.action}>
        <Text style={styles.swimmerDashboard}>{this.props.title}</Text>
      </View>
    );
  }
}

module.exports = SwimmerDashboard;
