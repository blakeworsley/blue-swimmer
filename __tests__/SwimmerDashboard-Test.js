import 'react-native';
import React from 'react';
import SwimmerDashboard from '../app/components/SwimmerDashboard';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <SwimmerDashboard />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a form on home page', () => {

})
