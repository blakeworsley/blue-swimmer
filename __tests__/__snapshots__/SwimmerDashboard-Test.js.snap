exports[`test renders correctly 1`] = `
<View
  style={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
      }
    } />
</View>
`;
